soc/nordic/nrf54h/soc.c

```diff
@@ -46,11 +76,32 @@ static void power_domain_init(void)
 	 *  WFI the power domain will be correctly retained.
 	 */
 
-	nrf_lrcconf_poweron_force_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_MAIN, true);
-	nrf_lrcconf_poweron_force_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_DOMAIN_0, true);
+	nrf_lrcconf_poweron_force_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_MAIN,
+			!IS_ENABLED(CONFIG_SOC_NRF54H20_CPURAD));
+	nrf_lrcconf_poweron_force_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_DOMAIN_0,
+			!IS_ENABLED(CONFIG_SOC_NRF54H20_CPURAD));
 
 	nrf_lrcconf_retain_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_MAIN, true);
 	nrf_lrcconf_retain_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_DOMAIN_0, true);
+	nrf_memconf_ramblock_ret_enable_set(NRF_MEMCONF, 0, RAMBLOCK_RET_BIT_ICACHE, false);
+	nrf_memconf_ramblock_ret_enable_set(NRF_MEMCONF, 0, RAMBLOCK_RET_BIT_DCACHE, false);
+	nrf_memconf_ramblock_ret_enable_set(NRF_MEMCONF, 1, RAMBLOCK_RET_BIT_ICACHE, false);
+	nrf_memconf_ramblock_ret_enable_set(NRF_MEMCONF, 1, RAMBLOCK_RET_BIT_DCACHE, false);
+#if defined(RAMBLOCK_RET2_BIT_ICACHE)
+	nrf_memconf_ramblock_ret2_enable_set(NRF_MEMCONF, 0, RAMBLOCK_RET2_BIT_ICACHE, false);
+	nrf_memconf_ramblock_ret2_enable_set(NRF_MEMCONF, 1, RAMBLOCK_RET2_BIT_ICACHE, false);
+#endif
+#if defined(RAMBLOCK_RET2_BIT_DCACHE)
+	nrf_memconf_ramblock_ret2_enable_set(NRF_MEMCONF, 0, RAMBLOCK_RET2_BIT_DCACHE, false);
+	nrf_memconf_ramblock_ret2_enable_set(NRF_MEMCONF, 1, RAMBLOCK_RET2_BIT_DCACHE, false);
+#endif
+	nrf_memconf_ramblock_ret_mask_enable_set(NRF_MEMCONF, 0, RAMBLOCK_RET_MASK, true);
+	nrf_memconf_ramblock_ret_mask_enable_set(NRF_MEMCONF, 1, RAMBLOCK_RET_MASK, true);
+#if defined(RAMBLOCK_RET2_MASK)
+	/* TODO: Use nrf_memconf_ramblock_ret2_mask_enable_set() function when it will be privided by HAL. **/
+	NRF_MEMCONF->POWER[0].RET2 |= RAMBLOCK_RET2_MASK;
```

Comment:
Would it be possible to do `NRF_MEMCONF->POWER[0].RET2 = predefined value` instead? `|=` results in read-modify-write that is inefficient when accessing peripherals' registers.

Reply:
Yep, but keep in mind that the `...mask_enable_set()` functions used above do it similarly : https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/hal/nrf_memconf.h#L376
Shall we introduce more efficient solution to HAL library?

Reply:
Definitely, we should. Or we'll spend ages unnecessarily reading register values.

But I guess that's out of the scope of this PR. Should we create a jira ticket to clean this up?

Reply:
Here it is: https://nordicsemi.atlassian.net/browse/NRFX-5862